api/routes/send_mail.route.js

```diff
@@ -0,0 +1,95 @@
+const express = require('express');
+const app = express();
+const mailRoutes = express.Router();
+//const multer = require('multer');
+//const { Readable } = require('stream');
+const mongodb = require('mongodb');
+const mongoose = require('mongoose');
+const MongoClient = require('mongodb').MongoClient;
+const ObjectID = require('mongodb').ObjectID;
+//const querystring = require('querystring');
+//const request = require('request');
+//const { parse, stringify } = require('node-html-parser');
+
+const mail = require('../mail');
+
+if(mail.couldNotCreate){
+  console.log("Could not create mail transporter which is required by the send_mail route. Refusing to continue.");
+  process.exit(1);
+}
+else {
+
+  // Mail transporter has been created successfully
+  // Continuing to create routes
+  const sendEmail = mail.sendEmail;
+
+  /*
+  const mailObj = {
+    from: "nrobinso@tcd.ie",
+    recipients: ["scealai.info@gmail.com"],
+    subject: "TEST 2 -- With passwords hidden -- This was send by nodejs",
+    message: "Hi everybody!",
+  };
+
+  sendEmail(mailObj).then((res) => {
+    console.log(res);
+  });
+  */
+
+
+  mailRoutes.route('/report_issue_anon').post(async (req, res) => {
+
+    console.log("reporting an issue from an anonymous user");
+    console.log(req.body);
+
+    let emailBody;
+    let subject = req.body.subject || "USER ISSUE REPORT --- ANONYMOUS";
+
+    // Make sure the client has sent a message in the body
+    if(req.body.message){
+      emailBody = req.body.message
+
+      // Refuse to send an email with just an empty string
+      if(emailBody === ""){
+        res.status(400).json({ error: "Message body empty. Refusing to send email."});
+        return;
+      }
+    }
+    else{
+      res.json({ error: "No message found in body of POST request to /mail/report_issue_anon. Refusing to send empty email."});
+      return;
+    }
```

Comment:
Nit: we can get rid of some of the nesting by checking `if (!...)`. This way we can return if true on the spot instead of handling the 'else' case somewhere else that's possibly far away in the file.

Also, it looks like in javascript `x === ""` is equivalent to `!x`, so I'm wondering if it's possible for the inner if statement ever to be called? See:
![image](https://user-images.githubusercontent.com/34962541/121737501-dafe6900-caf0-11eb-9459-200b0dd91ff0.png)


('nit' <--> 'this is kind of a personal preference so feel free to go with whatever reads better for you!')

```suggestion
if (!req.body.message) {
    res.json({ error: "No message found in body of POST request to /mail/report_issue_anon. Refusing to send empty email."});
    return;
}
emailBody = req.body.message

// Refuse to send an email with just an empty string
if (!emailBody) {
    res.status(400).json({ error: "Message body empty. Refusing to send email."});
    return;
}
```